source/faq.txt

```diff
@@ -6,12 +6,126 @@ FAQ
 
 .. default-domain:: mongodb
 
-Frequently Asked Questions
-
 .. contents:: On this page
    :local:
    :backlinks: none
    :depth: 1
    :class: singlecol
 
-Lorem ipsum.
+This page contains frequently asked questions and their corresponding answers.
+
+.. tip::
+
+   If you can't find an answer to your problem on this page,
+   see the :ref:`golang-issues-and-help` page for next steps and more
+   resources.
+   
+How Can I Fix the "WriteXXX can only write while positioned on a Element or Value but is positioned on a TopLevel" Error?
+-------------------------------------------------------------------------------------------------------------------------
+
+The ``bson.Marshal()`` method requires a parameter that can be decoded
+into a BSON document, such as the ``bson.D`` type. This error occurs
+when the input to ``bson.Marshal()`` is something *other* than a BSON
+document.
+
+Situations in which this error can occur include the following:
+
+- You pass a string to ``bson.Marshal()``, causing a ``WriteString`` error.
+- You pass a ``null`` to ``bson.Marshal()``, causing a ``WriteNull`` error.
+- You pass an integer to ``bson.Marshal()``, causing a ``WriteInt32`` error.
+
+You may encounter this error when you perform a CRUD operation that
+internally uses the ``bson.Marshal()`` method or when you call
+``bson.Marshal()`` directly to encode data.
+
+The following code produces a ``WriteNull`` error because the driver
+cannot encode the ``null`` valued ``sortOrder`` variable to BSON during
+the find-and-update operation:
+
+.. code-block:: go
+
+   var sortOrder bson.D
+   opts := options.FindOneAndUpdate().SetSort(sortOrder)
+   
+   update := bson.D{{"$inc", bson.D{{"counter", 1}}}}
```

Comment:
Similar comment to https://github.com/mongodb/docs-golang/pull/226#discussion_r1072938457

Reply:
changed to `updateDocument`